packages/ses/src/compartment-shim.js

```diff
@@ -100,16 +100,29 @@ export const CompartmentPrototype = {
       );
     }
 
-    return performEval(source, globalObject, localObject, {
-      globalTransforms,
-      localTransforms,
-      sloppyGlobalsMode,
-    });
+    return performEval(
+      source,
+      globalObject,
+      localObject,
+      {
+        globalTransforms,
+        localTransforms,
+        sloppyGlobalsMode,
+      },
+      scopeProxy => {
+        privateFields.get(this).scopeProxySet.add(scopeProxy);
+      },
+    );
   },
 
   toString() {
     return '[object Compartment]';
   },
+
+  /* eslint-disable-next-line no-underscore-dangle */
+  __isScopeProxy__(value) {
+    return privateFields.get(this).scopeProxySet.has(value);
+  },
```

Comment:
This does seem like a valid application of __ to indicate a feature that doesn’t telegraph an intent to standardize.